routing/bicycle_directions.cpp

```diff
@@ -222,39 +165,172 @@ Index::FeaturesLoaderGuard & BicycleDirectionsEngine::GetLoader(MwmSet::MwmId co
   return *m_loader;
 }
 
-void BicycleDirectionsEngine::LoadPathGeometry(UniNodeId const & uniNodeId,
-                                               vector<Junction> const & path,
-                                               LoadedPathSegment & pathSegment)
+void BicycleDirectionsEngine::LoadPathAttributes(FeatureID const & featureId, LoadedPathSegment & pathSegment)
 {
-  pathSegment.Clear();
-
-  if (!uniNodeId.GetFeature().IsValid())
-  {
-    ASSERT(false, ());
+  if (!featureId.IsValid())
     return;
-  }
 
   FeatureType ft;
-  if (!GetLoader(uniNodeId.GetFeature().m_mwmId)
-           .GetFeatureByIndex(uniNodeId.GetFeature().m_index, ft))
-  {
-    // The feature can't be read, therefore path geometry can't be
-    // loaded.
-    return;
-  }
+  GetLoader(featureId.m_mwmId).GetFeatureByIndex(featureId.m_index, ft);
 
   auto const highwayClass = ftypes::GetHighwayClass(ft);
   ASSERT_NOT_EQUAL(highwayClass, ftypes::HighwayClass::Error, ());
   ASSERT_NOT_EQUAL(highwayClass, ftypes::HighwayClass::Undefined, ());
 
   pathSegment.m_highwayClass = highwayClass;
   pathSegment.m_isLink = ftypes::IsLinkChecker::Instance()(ft);
-
   ft.GetName(FeatureType::DEFAULT_LANG, pathSegment.m_name);
-
-  pathSegment.m_nodeId = uniNodeId;
   pathSegment.m_onRoundabout = ftypes::IsRoundAboutChecker::Instance()(ft);
-  pathSegment.m_path = path;
-  // @TODO(bykoianko) It's better to fill pathSegment.m_weight.
+}
+
+void BicycleDirectionsEngine::GetUniNodeIdAndAdjacentEdges(
+    IRoadGraph::TEdgeVector const & outgoingEdges, FeatureID const & inFeatureId,
+    uint32_t startSegId, uint32_t endSegId, bool inIsForward, UniNodeId & uniNodeId,
+    BicycleDirectionsEngine::AdjacentEdges & adjacentEdges)
+{
+  // Outgoing edge angle is not used for bicycle routing.
+  adjacentEdges.m_outgoingTurns.isCandidatesAngleValid = false;
+  adjacentEdges.m_outgoingTurns.candidates.reserve(outgoingEdges.size());
+  uniNodeId = UniNodeId(inFeatureId, startSegId, endSegId, inIsForward);
+
+  for (auto const & edge : outgoingEdges)
+  {
+    auto const & outFeatureId = edge.GetFeatureId();
+    // Checking if |edge| is a fake edge.
+    if (!outFeatureId.IsValid())
+      continue;
+
+    FeatureType ft;
+    GetLoader(outFeatureId.m_mwmId).GetFeatureByIndex(outFeatureId.m_index, ft);
+
+    auto const highwayClass = ftypes::GetHighwayClass(ft);
+    ASSERT_NOT_EQUAL(highwayClass, ftypes::HighwayClass::Error, ());
+    ASSERT_NOT_EQUAL(highwayClass, ftypes::HighwayClass::Undefined, ());
+    adjacentEdges.m_outgoingTurns.candidates.emplace_back(0.0 /* angle */, uniNodeId, highwayClass);
```

Comment:
Yes, but later. Now |adjacentEdges.m_outgoingTurns.isCandidatesAngleValid = false;|. It mean the angles should not be taken into account while turn generation.